.github/actions/setup-nix/action.yml

```diff
@@ -0,0 +1,19 @@
+name: 'Setup Nix and Cachix'
+description: 'Installs Nix and configures Cachix for caching'
+inputs:
+  cachix-auth-token:
+    description: 'Authentication token for Cachix'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Install Nix
+      uses: DeterminateSystems/nix-installer-action@v20
+      with:
+        extra-conf: |
+          accept-flake-config = true
```

Comment:
NOTE: I though. It might be better to keep current inline style for all actions for different nix.conf. However it can be customized with inputs.